======================================
Test Cases for Use in 1-mul.py
======================================

Testing for function called "mul(a, b)".

"mul(a, b)" returns the product of two arguments, a and b. For numbers,
that value is equivalent to using the "*" operator.

     >>> mul = __import__('1-mul').mul

Exceptions
----------

Check for at least one argument to be float or int type.
      >>> mul("a", "b")
      Traceback (most recent call last):
      ...
      TypeError:  can't multiply sequence by non-int of type 'str'

Integer Cases
-----------

Check for both arguments as integers.
      >>> mul(2, 4)
      8

Check for one argument as float.
      >>> mul(1, 2.2)
      2.2

Check for one argument as None.
      >>> mul(1, None)
      Traceback (most recent call last):
      ...
      TypeError: unsupported operand type(s) for *: 'int' and 'NoneType'

Check for one argument as undefined.
      >>> mul(5, j)
      Traceback (most recent call last):
      ...
      NameError: name 'j' is not defined

Check for one argument as letter.
      >>> mul(5, "j")
      jjjjj

Check for one argument as forwardslash character (\).
      >>> mul(5, "\")
      SyntaxError: EOL while scanning string literal

Check for one argument as string.
      >>> mul(2, "Holberton")
      HolbertonHolberton

Check for one argument as dictionary.
      >>> mul(5, {'key': 'value'})
      Traceback (most recent call last):
      ...
      TypeError: unsupported operand type(s) for *: 'int' and 'dict'

Check for one argument as list.
      >>> mul(2, ['Hello', 8])
      ['Hello', 8, 'Hello', 8]

Check for one argument as tuple.
      >>> mul(3, (7, 8, 9))
      (7, 8, 9, 7, 8, 9, 7, 8, 9)

Float Case
----------

Check for both arguments as float.
      >>> mul(1.2, 2.3)
      2.76

Check for one argument as None.
      >>> mul(1.2, None)
      Traceback (most recent call last):
      ...
      TypeError: unsupported operand type(s) for *: 'float' and 'NoneType'

Check for one argument as undefined.
      >>>mul(5.7, help)
      Traceback (most recent call last):
      ...
      NameError: name 'Help' is not defined

Check for one argument as letter.
      >>>mul(2.5, 'n')
      Traceback (most recent call last):
      TypeError: can't multiply sequence by non-int of type 'float'

Check for one argument as string.
      >>>mul(2.5, "Error")
      Traceback (most recent call last):
      TypeError: can't multiply sequence by non-int of type 'float'

Check for one argument as dictionary.
      >>>mul(1.4, {"key": "value"})
      Traceback (most recent call last):
      TypeError: unsupported operand type(s) for *: 'float' and 'dict'

Check for one argument as list.
      >>>mul(3.7, [1, 2, 3])
      Traceback (most recent call last):
      TypeError: can't multiply sequence by non-int of type 'float'

Check for one argument as tuple.
      >>>mul(9.8, (1, 3, 5, 7, 9))
      Traceback (most recent call last):
      TypeError: can't multiply sequence by non-int of type 'float'
